Android App/app/src/main/java/org/cfp/citizenconnect/SplashScreen.java

```diff
@@ -74,30 +74,29 @@ public boolean isObjectExist() {
     }
 
     private class DownloadFilesTask extends AsyncTask<Void, Integer, Void> {
-        ProgressBar progress;
+
         Context context;
 
-        public DownloadFilesTask(ProgressBar progress, Context context) {
-            this.progress = progress;
+        public DownloadFilesTask(Context context) {
+
             this.context = context;
 
         }
 
         @Override
         protected void onPreExecute() {
-            progress.setVisibility(ProgressBar.VISIBLE);
         }
 
         @Override
         protected Void doInBackground(Void... voids) {
 
             getDataSet(response -> {
-                progress.setVisibility(ProgressBar.GONE);
+
                 startActivity(new Intent(SplashScreen.this, MainActivity.class));
```

Comment:
Is there any particular reason why you are starting `SplashScreen` again from the same activity and that too in both cases of success or error?

Reply:
@shahzaib414 @wahibhaq lets discuss it

Reply:
As i see i am not starting SplashScreen, i am starting MainActivity.

Reply:
Ok I must have overlooked it. Thanks for correcting me.

But can you share why it's going forward to MainScreen even when there is an error? Also if you think it's necessary to launch at several locations then it might be better to place it in a function launchMainScreen() and call this. Less code is always better 🙂

Reply:
there is fragment in App Data Set, I am retrieving data from firebase here and storing to local storage to use later , So according to Application requirement, it's not necessary that if App fails to retrieve data from firebase it shouldn't continue. Because i have handled this case in App  while loading data lists . that's why i am redirecting user to main Activity in both cases. Better suggestions to do this would be appreciated . I hope i have cleared your point .

Reply:
Hmm ok. I would like to invest some time to understand the flow as I don't have a solid understanding of the underlying implementation. I will get back to you if things could be done in a better way 👍